gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/PrimaryOwnerServiceImpl.java

```diff
@@ -58,77 +56,35 @@ public class PrimaryOwnerServiceImpl extends TransactionalService implements Pri
     private final GroupService groupService;
     private final ParameterService parameterService;
     private final RoleService roleService;
+    private final ApiPrimaryOwnerDomainService primaryOwnerDomainService;
 
     public PrimaryOwnerServiceImpl(
         final UserService userService,
         final MembershipService membershipService,
         final GroupService groupService,
         final ParameterService parameterService,
-        final RoleService roleService
+        final RoleService roleService,
+        ApiPrimaryOwnerDomainService primaryOwnerDomainService
     ) {
         this.userService = userService;
         this.membershipService = membershipService;
         this.groupService = groupService;
         this.parameterService = parameterService;
         this.roleService = roleService;
+        this.primaryOwnerDomainService = primaryOwnerDomainService;
     }
 
     @Override
     public PrimaryOwnerEntity getPrimaryOwner(final ExecutionContext executionContext, final String apiId)
         throws TechnicalManagementException {
-        MembershipEntity primaryOwnerMemberEntity = membershipService.getPrimaryOwner(
-            executionContext.getOrganizationId(),
-            io.gravitee.rest.api.model.MembershipReferenceType.API,
-            apiId
-        );
-        if (primaryOwnerMemberEntity == null) {
-            throw new PrimaryOwnerNotFoundException(apiId);
-        }
-        if (MembershipMemberType.GROUP == primaryOwnerMemberEntity.getMemberType()) {
-            final String poMail = computePrimaryOwnerMailFromGroup(executionContext, primaryOwnerMemberEntity.getMemberId());
-            return new PrimaryOwnerEntity(groupService.findById(executionContext, primaryOwnerMemberEntity.getMemberId()), poMail);
-        }
-        return new PrimaryOwnerEntity(userService.findById(executionContext, primaryOwnerMemberEntity.getMemberId()));
+        var po = primaryOwnerDomainService.findApiPrimaryOwner(executionContext, apiId);
+        return PrimaryOwnerEntity.builder().id(po.getId()).type(po.getType()).displayName(po.getDisplayName()).email(po.getEmail()).build();
```

Comment:
Here I would use an Adapter provided by the new architecture for the legacy one.

We should avoid this "in-legacy-code" mapping because we could lose information if model evolve on one side

wdyt ?

Reply:
We can not create a static method to build new PrimaryOwnerEntity type from the old one because model project doesn't have access to service. So I created a static method in the class to do the mapping.